ocaml/src/alba_client.ml

```diff
@@ -213,6 +213,33 @@ class alba_client (base_client : Alba_base_client.client)
     >>= fun _len ->
     Lwt.return !res
 
+  method download_object_to_bytes
+           ~namespace
+           ~object_name
+           ~consistent_read
+           ~should_cache
+         : Lwt_bytes.t option Lwt.t =
+    let res = ref None in
```

Comment:
In this scope we don't yet know how big the Lwt_bytes.t needs to be, so we can't create it yet.
We need the reference here so we can return it later on (having it only available in write_object_data below is not enough).

It does look like I'm introducing a potential space/memory leak here (in case sth goes wrong during download_object_generic). I'll fix that.